### DISCUSSION TITLE
GLIBC_2.33 & GLIBC_2.34

### DISCUSSION THREAD

**Discussion by dcipher** [webui-dev/webui#358] [Q&A]
I have been using webui on MS Windows but when I try to use it on OpenSuse I get linker errors saying the webui-2.so requires glibc_2.33 and glibc_2.34 but they are not available for OpenSuse Leap 15.5.  So my question is what Linux version is recommended for webui-2.so?

Thanks,
Simon

**Comment by wwderw** [comment]
When it goes for Linux, I tend to build against musl.  That precludes dealing with much of this, depending on what other libs one is using as well, but if just doing straight webui, it builds fine.

**Comment by AlbertShown** [comment]
This is a [known issue](https://github.com/webui-dev/webui/pull/357), get fixed days ago in the [Nightly Build](https://github.com/webui-dev/webui/releases/tag/nightly).

Solution 1: Simply re-download the latest [Nightly Build](https://github.com/webui-dev/webui/releases/tag/nightly).
Solution 2: Re-build from source:
```
git clone https://github.com/webui-dev/webui.git
cd webui
make
```

**Reply by dcipher** [reply]
Hi

I checked the nightly build and it is not fixed.  It was dated May 9, 2024.

remote@MPLiteTest:~/Downloads> ldd webui-2.so
./webui-2.so: /lib64/libc.so.6: version `GLIBC_2.33' not found (required by ./webui-2.so)
./webui-2.so: /lib64/libc.so.6: version `GLIBC_2.34' not found (required by ./webui-2.so)
	linux-vdso.so.1 (0x00007ffd7b582000)
	libc.so.6 => /lib64/libc.so.6 (0x00007f680c30c000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f680c56a000)

**Reply by dcipher** [reply]
I should add that I was able to build it on my OpenSuse 15.5 without any issues. So now I do not have to use the pre-built binaries.

**Reply by AlbertShown** [reply]
Thank you for confirming this 👍

Yes, The nightly build is using GitHub `Ubuntu 20.4`, wish is using `GLIBC_2.33`.
So, to use webui with old GLIBC version, the user should re-build webui from source.
